From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Bobbe <cbobbe@zulip.com>
Date: Tue, 10 Jan 2023 10:40:01 -0800
Subject: [tsflower] Fill in simple types from @types/react-native; rewrite TS
 `Pick`

For more discussion of how to translate TS's `Pick` helper, see a
previous patch that manually translated it to SubsetProperties in
the @react-navigation types.
---
 .../lib/typescript/index.js.flow                | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git types/react-native-document-picker/lib/typescript/index.js.flow types/react-native-document-picker/lib/typescript/index.js.flow
index e73492740..c605dae3e 100644
--- types/react-native-document-picker/lib/typescript/index.js.flow
+++ types/react-native-document-picker/lib/typescript/index.js.flow
@@ -2,7 +2,6 @@
  * @generated by TsFlower
  */
 import type { Readonly } from 'tsflower/subst/lib';
-import { type ModalPropsIOS } from 'react-native';
 import { type PlatformTypes, type SupportedPlatforms } from './fileTypes';
 
 export type DocumentPickerResponse = {
@@ -80,12 +79,20 @@ export type DocumentPickerOptions<OS: SupportedPlatforms> = {
   copyTo?: 'cachesDirectory' | 'documentDirectory',
   allowMultiSelection?: boolean,
   transitionStyle?: TransitionStyle,
+
+  // TODO(tsflower): Add ModalPropsIOS to tsflower/subst/react-native and
+  //   add macro to rewrite `Pick`.
+  presentationStyle?: 'fullScreen' | 'pageSheet' | 'formSheet' | 'overFullScreen' | void,
   ...
-} & Pick<ModalPropsIOS, 'presentationStyle'>;
+};
 
-declare export function pickDirectory<OS: SupportedPlatforms>(
-  params?: Pick<DocumentPickerOptions<OS>, 'presentationStyle' | 'transitionStyle'>,
-): Promise<DirectoryPickerResponse | null>;
+declare export function pickDirectory<OS: SupportedPlatforms>(params?: {
+  // TODO(tsflower): Add ModalPropsIOS to tsflower/subst/react-native and
+  //   add macro to rewrite `Pick`.
+  presentationStyle?: 'fullScreen' | 'pageSheet' | 'formSheet' | 'overFullScreen' | void,
+  transitionStyle?: DocumentPickerOptions<OS>['transitionStyle'],
+  ...
+}): Promise<DirectoryPickerResponse | null>;
 
 declare export function pickMultiple<OS: SupportedPlatforms>(
   opts?: DocumentPickerOptions<OS>,
-- 
2.32.0

